apis/securesourcemanager/v1alpha1/securesourcemanagerrepository_types.go

```diff
@@ -0,0 +1,111 @@
+// Copyright 2024 Google LLC
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package v1alpha1
+
+import (
+	refs "github.com/GoogleCloudPlatform/k8s-config-connector/apis/refs/v1beta1"
+	"github.com/GoogleCloudPlatform/k8s-config-connector/pkg/apis/k8s/v1alpha1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+var SecureSourceManagerRepositoryGVK = GroupVersion.WithKind("SecureSourceManagerRepository")
+
+// EDIT THIS FILE!  THIS IS SCAFFOLDING FOR YOU TO OWN!
+// NOTE: json tags are required.  Any new fields you add must have json tags for the fields to be serialized.
+
+// SecureSourceManagerRepositorySpec defines the desired state of SecureSourceManagerRepository
+// +kcc:proto=google.cloud.securesourcemanager.v1.Repository
+type SecureSourceManagerRepositorySpec struct {
+	/* Immutable. The Project that this resource belongs to. */
+	// +required
+	ProjectRef *refs.ProjectRef `json:"projectRef"`
+
+	/* Immutable. Location of the instance. */
+	Location string `json:"location"`
```

Comment:
Is location required or optional?

I'm asking because we found the CRD generator will mark location as required by default(we'll need to address this in the future): https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/835aeccf17385962c9d0f720719207fa42d78952/config/crds/resources/apiextensions.k8s.io_v1_customresourcedefinition_securesourcemanagerrepositories.securesourcemanager.cnrm.cloud.google.com.yaml#L279

If this is required, add +required tag to make it clear; If it's optional, we need to explicitly add +optional tag to avoid it's default to required in the CRD.

Reply:
Required, thanks. I've updated the `location` field in `securesourcemanagerinstance_types.go` too. 

I have two more questions.

Should `ResourceID` be annotated as well? 

Should this `// +kubebuilder:validation:XValidation:rule="self == oldSelf",message="ResourceID field is immutable"` annotation be added to all immutable fields? I don't really understand the difference between option 1 and 2 here https://github.com/GoogleCloudPlatform/k8s-config-connector/blob/master/docs/develop-resources/api-conventions/validations.md#option-1

Reply:
So instance does not have `(google.api.field_behavior) = IMMUTABLE`([code](https://source.corp.google.com/piper///depot/google3/google/cloud/securesourcemanager/v1/secure_source_manager.proto;l=1737)), so we that annotation

Reply:
Removed immutable from the instance ref